### PR TITLE
Supervisor create course

### PR DESCRIPTION
**Pull request by trantube94**
I have created a pull request, check it please!

### CODE REVIEW

File: app/decorators/supervisor/course_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+class Supervisor::CourseDecorator < Draper::Decorator
+  delegate_all
+end
```
Comment (VuongLong) [inline app/decorators/supervisor/course_decorator.rb]: Remove this file

**Comment by minhld1709** [comment]
OK

**Comment by VuongLong** [comment]
OK

File: app/controllers/supervisor/courses_controller.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      redirect_to new_supervisor_course_path
+    end
+  end
+
+  private
+  def course_params
+    params.require(:course).permit :name, :description, subject_ids: []
+  end
+
+  def init_subjects
```
Comment (dieunb) [inline app/controllers/supervisor/courses_controller.rb]: function name is not good, suggest use `load_subjects`

File: app/controllers/supervisor/courses_controller.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  before_action :init_subjects, only: :new
+
+  def new
+    @course = Course.new
+  end
+
+  def create
+    @course = Course.new course_params
+    if @course.save
+      flash[:success] = t "supervisor.course.create_success"
```
Comment (dieunb) [inline app/controllers/supervisor/courses_controller.rb]: Should create general flash for locale.

File: app/views/supervisor/courses/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<%= form_for [:supervisor, @course] do |f| %>
+  <%= f.label :name %>
+  <%= f.text_field :name %>
+
+  <%= f.label :description %>
+  <%= f.text_area :description %>
+
+  <% @subjects.each do |subject| %>
+    <%= f.label :subject_ids, subject.name %>
+    <%= check_box_tag :subject_ids, subject.id,
```
Comment (dieunb) [inline app/views/supervisor/courses/new.html.erb]: Why don't use `f` builder?

File: app/views/supervisor/courses/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,15 @@
+<%= form_for [:supervisor, @course] do |f| %>
+  <%= f.label :name %>
+  <%= f.text_field :name %>
+
+  <%= f.label :description %>
+  <%= f.text_area :description %>
+
+  <% @subjects.each do |subject| %>
+    <%= f.label :subject_ids, subject.name %>
+    <%= f.check_box :subject_ids, {name: "course[subject_ids][]",
```
Comment (dieunb) [inline app/views/supervisor/courses/new.html.erb]: Có cần set name nữa không em?

File: config/locales/en.yml [hunk 6]
```diff
@@ -1,23 +1,6 @@
...
-#
-# To learn more, please read the Rails Internationalization guide
-# available at http://guides.rubyonrails.org/i18n.html.
-
 en:
-  hello: "Hello world"
+  action:
+    create: Create
+  flash:
+    create_success: Create success
```
Comment (dieunb) [inline config/locales/en.yml]: Message in ra ghi rõ ràng và đầy đủ hơn nhé

File: app/controllers/supervisor/courses_controller.rb [hunk 7]
```diff
@@ -0,0 +1,27 @@
+class Supervisor::CoursesController < ApplicationController
+  before_action :load_all_subjects, only: :new
+
+  def new
+    @course = Course.new
```
Comment (hellovietnam93) [inline app/controllers/supervisor/courses_controller.rb]: remove this line. Using cancancan

File: config/routes.rb [hunk 8]
```diff
@@ -3,6 +3,11 @@
   mount RailsAdmin::Engine => "/admin", as: "rails_admin"
   root "static_pages#home"
   devise_for :users, path: "auth", path_names: {sign_in: "login", sign_out: "logout"}
+
+  namespace :supervisor do
```
Comment (hellovietnam93) [inline config/routes.rb]: remove namespace

**Pull request closed without merging**